### ISSUE TITLE
Enum - Value is unsubscriptable

### ISSUE THREAD

**Issue by kemar** [PyCQA/pylint#992]
### Steps to reproduce

``` bash
python -V
Python 3.5.2
```

``` python
import enum

TASK_STATUS = enum.Enum('Task', (('todo', 1), ('success', 2)))

status = TASK_STATUS['success']
```
### Current behavior

```
msg_id E1136
symbol unsubscriptable-object
obj Value 'TASK_STATUS'
msg is unsubscriptable
```
### Expected behavior

No warning.

This is [the right way for programmatic access to enumeration members](https://docs.python.org/3/library/enum.html#programmatic-access-to-enumeration-members-and-their-attributes).
### pylint --version output

``` bash
pylint --version
No config file found, using default configuration
pylint 1.6.1, 
astroid 1.4.7
Python 3.5.2 (default, Jun 27 2016, 03:10:38) 
[GCC 4.2.1 Compatible Apple LLVM 7.0.2 (clang-700.1.81)]
```

**Comment by PCManticore** [comment]
Thanks! We have definitely some issues with understanding whatever Enum is doing, so looking forward to having these issues fixed.

**Comment by hoootsli** [comment]
First of all, thank you very much for all the great work on pylint! :)

In the context of this issue: It looks like [the recommended way of iterating the members of an `Enum`](https://docs.python.org/3/library/enum.html#iteration) also leads to a pylint warning `... has no '__members__' member`. Will that issue be addressed at the same time? (I tested with pylint 1.6.4.)
